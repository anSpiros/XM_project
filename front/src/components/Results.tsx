import React from "react";
import PropTypes from "prop-types";
import { Bar } from "react-chartjs-2";
import { DisplayData } from "../interfaces/results.interface";
import DisplayDataTable from "./DisplayDataTable";

Results.propTypes = {
  chartData: PropTypes.object.isRequired,
  chartOptions: PropTypes.object.isRequired,
  displayDataForTable: PropTypes.object.isRequired,
};

interface ResultsProps {
  chartData: any;
  chartOptions: any;
  companyDescription: string;
  displayDataForTable: DisplayData;
}

function Results({
  chartData,
  chartOptions,
  companyDescription,
  displayDataForTable,
}: ResultsProps) {
  return (
    <>
      <div className="row results">
        <div className="col-12 results-company">
          <h3 className="results-company__name">{companyDescription}</h3>
        </div>
        <div className="col-12">
          <div className="chart-wrapper">
            <Bar data={chartData} options={chartOptions} />
          </div>
        </div>
        <div className="col-12 results-table">
          <DisplayDataTable
            bodyData={displayDataForTable.body}
            headerData={displayDataForTable.header}
          />
        </div>
      </div>
    </>
  );
}

export default Results;
