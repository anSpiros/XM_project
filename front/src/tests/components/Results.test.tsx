import React from "react";
import { render } from "@testing-library/react";
import Results from "../../components/Results";

const chartData = {
  labels: ["test", "test2"],
  datasets: [
    {
      label: "open",
      data: [1, 2],
      backgroundColor: "test",
      borderWidth: 4,
    },
    {
      label: "close",
      data: [3, 4],
      backgroundColor: "test",
      borderWidth: 4,
    },
  ],
};

const displayDataForTable = {
  body: [
    {
      date: "test",
      open: 1,
      high: 2,
      low: 1,
      close: 2,
      volume: 3,
    },
  ],
  header: [
    {
      column: "Date",
    },
    {
      column: "Open",
    },
    {
      column: "High",
    },
    {
      column: "Low",
    },
    {
      column: "Close",
    },
    {
      column: "Volume",
    },
  ],
};

const chartOptions = {
  responsive: true,
  scales: {
    yAxes: [
      {
        ticks: {
          beginAtZero: true,
          stacked: true,
        },
        gridLines: {
          display: false,
        },
      },
    ],
    xAxes: [
      {
        ticks: {
          stacked: true,
        },
        gridLines: {
          display: false,
        },
      },
    ],
  },
};

jest.mock("react-chartjs-2", () => ({
  Bar: () => null,
}));

describe("Results component", () => {
  test("should render correctly", () => {
    const { asFragment } = render(
      <Results
        chartData={chartData}
        chartOptions={chartOptions}
        companyDescription={"test"}
        displayDataForTable={displayDataForTable}
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
