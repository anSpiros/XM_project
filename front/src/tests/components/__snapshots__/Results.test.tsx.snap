// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Results component should render correctly 1`] = `
<DocumentFragment>
  <div
    class="row results"
  >
    <div
      class="col-12 results-company"
    >
      <h3
        class="results-company__name"
      >
        test
      </h3>
    </div>
    <div
      class="col-12"
    >
      <div
        class="chart-wrapper"
      />
    </div>
    <div
      class="col-12 results-table"
    >
      <div
        class="table-responsive"
      >
        <table
          class="table"
        >
          <thead>
            <tr>
              <th>
                Date
              </th>
              <th>
                Open
              </th>
              <th>
                High
              </th>
              <th>
                Low
              </th>
              <th>
                Close
              </th>
              <th>
                Volume
              </th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td>
                test
              </td>
              <td>
                1
              </td>
              <td>
                2
              </td>
              <td>
                1
              </td>
              <td>
                2
              </td>
              <td>
                3
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
